### PR TITLE
HFI Calc: Better error handling

### PR DESCRIPTION
**Pull request by andrea-williams**
- Fixed missing error icon in Prep Level cell
# Test Links:
[Percentile Calculator](https://wps-pr-1966.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1966.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1966.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1966.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1966.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN)
[Fire Behaviour Advisory](https://wps-pr-1966.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1966.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-1966.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: api/app/hfi/hfi_calc.py [hunk 1]
```diff
@@ -308,18 +308,20 @@ def calculate_daily_results(num_prep_days: int,
...
     for index in range(num_prep_days):
         dailies_date = start_date + timedelta(days=index)
         prep_day_dailies = get_prep_day_dailies(dailies_date, area_dailies)
         daily_fire_starts: FireStartRange = planning_area_fire_starts[area_id][index]
-        mean_intensity_group = calculate_mean_intensity(prep_day_dailies)
+        mean_intensity_group = calculate_mean_intensity(prep_day_dailies, num_unique_station_codes)
         prep_level = calculate_prep_level(mean_intensity_group, daily_fire_starts, fire_start_lookup)
         validated_dailies: List[ValidatedStationDaily] = list(map(validate_station_daily, prep_day_dailies))
         # check if all validated_dailies are valid.
-        all_dailies_valid = all(map(lambda validated_daily: (validated_daily.valid), validated_dailies))
```
Comment (andrea-williams) [inline api/app/hfi/hfi_calc.py:322]: something about this wasn't working - it kept marking everything as valid always.

**Review by tessmgood — CHANGES_REQUESTED** [review]
Looking much better, still seeing a few things: 

When I load Kamloops: May 4-8 there are errors in the daily Prep columns, but the final calculated Prep column is empty. It should also be displaying an error. 

Same FC/Date: When I got to the May 4 daily tab to investigate errors, I scroll down to Vernon planning area, and exclude Kettle 2 (error), it still can't calculate prep because of missing Fire Weather Indices for Turtle. This is unclear 
- From memory (please correct me!) we discussed putting an error in the final FWI column when this happens, to show something is wrong, but not to be overwhelming and display 6 error icons in a row from FFMC through FWI 
  - that said, the pattern for incomplete weather parameters is to display errors in each cell that is missing data, we could just throw 6 errors in if it's faster

**Review by Sybrand — APPROVED** [review]

**Review by conbrad — APPROVED** [review]
Nice work! Approving assuming Tess' feedback is addressed.

File: api/app/hfi/hfi_calc.py (hunk 1 above)
Comment (conbrad) [reply api/app/hfi/hfi_calc.py:322]: Weird, maybe the `map` needed to be wrapped in a `list`? Perhaps `all` doesn't know how to handle a wrap `map` iterable object.

**Review by tessmgood — APPROVED** [review]

File: web/src/features/hfiCalculator/components/DailyViewTable.tsx [hunk 2]
```diff
@@ -355,90 +355,81 @@ export const DailyViewTable = (props: Props): JSX.Element => {
...
-                          daily={daily}
-                          errorToolTipText={
-                            'FWI cannot be null. Impacts ROS, HFI, Fire Type, FIG, Prep calculations.'
-                          }
-                        />
+                          ] as KeyTooltipDecimalPlacesTuple,
+                          [
+                            'ffmc' as keyof StationDaily,
+                            'FFMC cannot be null. Impacts ROS, HFI, Fire Type, FIG, Prep calculations.',
+                            0
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/DailyViewTable.tsx]: Seems like none of these parameters differ, can we just use a constant?

Comment (andrea-williams) [reply web/src/features/hfiCalculator/components/DailyViewTable.tsx]: The params do differ somewhat.

Comment (conbrad) [reply web/src/features/hfiCalculator/components/DailyViewTable.tsx]: Does `0` differ?

Comment (andrea-williams) [reply web/src/features/hfiCalculator/components/DailyViewTable.tsx]: I think you might be looking at an older version - try refreshing the page. I've removed the 0s for now (they might return later if/when we decide to remove decimal points from some of our values)

Comment (andrea-williams) [reply web/src/features/hfiCalculator/components/DailyViewTable.tsx]: Sorry, when you said none of the params differ, I thought you were talking about the contents of error message tooltips.

Comment (conbrad) [reply web/src/features/hfiCalculator/components/DailyViewTable.tsx]: Nope, my bad, I should have specified which parameter. LGTM.

**Pull request merged**